guardian/elastic-search-monitor
# Upgrade to latest GuCDK version (49.1.0)

## What does this change?

The 2 most significant GuCDK changes for this project are:

1. GuCDK has upgraded the `aws-cdk` dependencies from v1 to v2. Projects which use GuCDK to define their infrastructure have to match the versions for these `aws-cdk` libraries. Many of the changes in this PR therefore relate to that upgrade. See the [GuCDK release notes for v41.0.0](https://github.com/guardian/cdk/releases/tag/v41.0.0) for more details.
2. The `GuSnsTopic` construct has been removed. We have moved to a `Topic` (this is an AWS construct, rather than a Guardian construct). This should have not modify the underlying infrastructure as the logical ID is unchanged. See the [GuCDK release notes for v49.0.0](https://github.com/guardian/cdk/releases/tag/v49.0.0) for more details.

## How to test

The snapshot tests show that we are adding metadata (for tracking) and tightening up permissions. I have also previewed the change set to confirm that no resources are being replaced:

![image](https://user-images.githubusercontent.com/19384074/216946669-019744fe-2017-41e9-b893-68986433de63.png)

All of these modifications are expected as GuCDK upgrades update the tags on every resource.

## How can we measure success?

We'll be using an up to date version of GuCDK, which makes it easier to adopt new features!

## Have we considered potential risks?

Yes, we are upgrading several major versions here so it's possible that I've missed something. I will check logs etc. after this PR is merged to confirm that everything is still working as expected.
